src/uct/api/uct.h

```diff
@@ -1213,6 +1215,72 @@ struct uct_md_attr {
 };
 
 
+/**
+ * @ingroup UCT_MD
+ * @brief UCT MD memory attributes field mask
+ *
+ * The enumeration allows specifying which fields in @ref uct_md_mem_attr_t
+ * are present.
+ */
+enum uct_md_mem_attr_field {
+    UCT_MD_MEM_ATTR_FIELD_MEM_TYPE = UCS_BIT(0), /**< Indicate if memory type
+                                                      is populated. Eg: CPU/GPU */
+    UCT_MD_MEM_ATTR_FIELD_SYS_DEV  = UCS_BIT(1)  /**< Indicate if details of
+                                                      system device backing
+                                                      the pointer is populated.
+                                                      Eg: NUMA, GPU, etc */
+};
+
+
+/**
+ * @ingroup UCT_MD
+ * @brief  Memory domain attributes.
+ *
+ * This structure defines the attributes of a memory pointer which may
+ * include memory type of the pointer, the system device that backs the
+ * pointer depending on the bit fields populated in field_mask.
+ */
+typedef struct uct_md_mem_attr {
+    /**
+     * Mask of valid fields in this structure, using bits from
+     * @ref uct_md_mem_attr_t. Note that the field mask is
+     * populated upon return from uct_md_mem_query and not set by user.
+     * Subsequent use of members of the structure are valid after ensuring that
+     * relevant bits in the field_mask are set.
+     */
+    uint64_t          field_mask;
+
+    /**
+     * Is the type CPU memory or GPU memory, etc
+     */
+    ucs_memory_type_t mem_type;
+
+    /**
+     * Location of device pointer. eg: NUMA/GPU
+     */
+    ucs_sys_device_t  sys_dev;
+} uct_md_mem_attr_t;
+
+
+/**
+ * @ingroup UCT_MD
+ * @brief Query attributes of a given pointer
+ *
+ * Return attributes such as memory type, and system device for the
+ * given pointer of specific length.
+ *
+ * @param [in]     md          Memory domain to allocate memory on.
```

Comment:
I'd rephrase somehow to reflect that this function does not allocate memory. E.g, like below:
`Memory domain the memory was allocated on`?

Reply:
Thanks for the catch @brminich

This was a copy paste error. I'll rephrase this

Reply:
@brminich 
I want to avoid `MD where memory was allocated on` because if cuda pointers are queried with this function, then it's not necessary that UCT cuda md was used to allocate. I think the same argument applies to host pointers. The query function would return failure for pointer queries on MDs that don't recognize the pointer kind.

Does the following work instead?
```
 * @param [in]     md          Memory domain to run query on. 
                                           This function returns error if MD does not recognize the pointer.
```

Reply:
👍

Reply:
@Akshay-Venkatesh, looks good to me. Thanks